server/cluster/metadata/cluster_metadata.go

```diff
@@ -752,3 +754,20 @@ L1:
 
 	return true
 }
+
+func (c *ClusterMetadata) maybeCorrectShardVersion(ctx context.Context, node RegisteredNode) {
+	topology := c.topologyManager.GetTopology()
+	for _, shardInfo := range node.ShardInfos {
+		oldShardView := topology.ShardViewsMapping[shardInfo.ID]
```

Comment:
Shall we check whether the oldShardView is found in the mapping? Otherwise it will panic at the next line.